File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario3/id_108T2000Test.py
# id_108_Test.java
from scenario3.id_108 import count_nums

import pytest

class id_108_Test:
    """
    Test class of id_108.py.
    It contains ten unit test cases for the :func:`scenario3.id_108.count_nums`.
    """


    def test_count_nums_1(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([]) == 0

    def test_count_nums_2(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, 11, -11]) == 1

    def test_count_nums_3(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([1, 1, 2]) == 3

    def test_count_nums_4(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2]) == 0

    def test_count_nums_5(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2]) == 0

    def test_count_nums_6(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3]) == 0

    def test_count_nums_7(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3]) == 0

    def test_count_nums_8(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3, -4]) == 0

    def test_count_nums_9(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3, -4, -4]) == 0

    def test_count_nums_10(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3, -4, -4, -5]) == 0

# id_108_Test.py
from scenario3.id_108 import count_nums

import pytest

class id_108_Test:
    """
    Test class of id_108.py.
    It contains ten unit test cases for the :func:`scenario3.id_108.count_nums`.
    """
		
    def test_count_nums_1(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([]) == 0

    def test_count_nums_2(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, 11, -11]) == 1

    def test_count_nums_3(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([1, 1, 2]) == 3

    def test_count_nums_4(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2]) == 0

    def test_count_nums_5(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2]) == 0

    def test_count_nums_6(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3]) == 0

    def test_count_nums_7(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3]) == 0

    def test_count_nums_8(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3, -4]) == 0

    def test_count_nums_9(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3, -4, -4]) == 0

    def test_count_nums_10(self):
        """
        Test case for the function :func:`scenario3.id_108.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2, -3, -3, -4, -4, -5]) == 0

# id_109.py

def count_nums(arr):
    """
    Write a function count_nums which takes an array of integers and returns
    the number of elements which has a sum of digits > 0.
    If a number is negative, then its first signed digit will be negative:
    e.g. -123 has signed digits -1, 2, and 3.
    >>> count_nums([]) == 0
    >>> count_nums([-1, 11, -11]) == 1
    >>> count_nums([1, 1, 2]) == 3
    """

    pass
# id_109_Test.java
from scenario3.id_109 import count_nums

import pytest

class id_109_Test:
    """
    Test class of id_109.py.
    It contains ten unit test cases for the :func:`scenario3.id_109.count_nums`.
    """
		
    def test_count_nums_1(self):
        """
        Test case for the function :func:`scenario3.id_109.count_nums`.
        """
        assert count_nums([]) == 0

    def test_count_nums_2(self):
        """
        Test case for the function :func:`scenario3.id_109.count_nums`.
        """
        assert count_nums([-1, 11, -11]) == 1

    def test_count_nums_3(self):
        """
        Test case for the function :func:`scenario3.id_109.count_nums`.
        """
        assert count_nums([1, 1, 2]) == 3

    def test_count_nums_4(self):
        """
        Test case for the function :func:`scenario3.id_109.count_nums`.
        """
        assert count_nums([-1, -1, -2]) == 0

    def test_count_nums_5(self):
        """
        Test case for the function :func:`scenario3.id_109.count_nums`.
        """
        assert count_nums([-1, -1, -2, -2]) == 0

    def test_count_nums_6(self):
        """
        Test case for the function :

File: RQ2_Prompt_Elements/GPT3.5_Data/HumanEvalJava_output/scenario3/id_108T2000Test.java
// CountNums.java
package scenario3;

import java.util.List;

class CountNums {

    /**
     * * Write a function count_nums which takes an array of integers and returns
     * the number of elements which has a sum of digits > 0.
     * If a number is negative, then its first signed digit will be negative:
     * e.g. -123 has signed digits -1, 2, and 3.
     * >>> countNums([]) == 0
     * >>> countNums([-1, 11, -11]) == 1
     * >>> countNums([1, 1, 2]) == 3
     *
     * > countNums([])
     * 0
     * > countNums([-1, -2, 0])
     * 0
     * > countNums([1, 1, 2, -2, 3, 4, 5])
     * 6
     * > countNums([1, 6, 9, -6, 0, 1, 5])
     * 5
     * > countNums([1, 100, 98, -7, 1, -1])
     * 4
     * > countNums([12, 23, 34, -45, -56, 0])
     * 5
     * > countNums([0, 1])
     * 1
     * > countNums([1])
     * 1
     */
    public static int countNums(List<Integer> arr) {
        int count = 0;
        for (int num : arr) {
            int sum = 0;
            int temp = num;
            while (temp != 0) {
                sum += temp % 10;
                temp /= 10;
            }
            if (num < 0) {
                sum *= -1;
            }
            if (sum != 0) {
                count++;
            }
        }
        return count;
    }
}

// CountNumsTest.java
package scenario3;

import java.util.Arrays;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link CountNums}.
* It contains ten unit test cases for the {@link CountNums#countNums(List)} method.
*/
class CountNumsTest {
    
    @Test
    void testEmptyList() {
        List<Integer> arr = Arrays.asList();
        assertEquals(0, CountNums.countNums(arr));
    }
    
    @Test
    void testAllZeros() {
        List<Integer> arr = Arrays.asList(0, 0, 0);
        assertEquals(0, CountNums.countNums(arr));
    }
    
    @Test
    void testAllNegatives() {
        List<Integer> arr = Arrays.asList(-1, -2, -3);
        assertEquals(0, CountNums.countNums(arr));
    }
    
    @Test
    void testOnePositive() {
        List<Integer> arr = Arrays.asList(1);
        assertEquals(1, CountNums.countNums(arr));
    }
    
    @Test
    void testOneNegative() {
        List<Integer> arr = Arrays.asList(-1);
        assertEquals(1, CountNums.countNums(arr));
    }
    
    @Test
    void testMixed() {
        List<Integer> arr = Arrays.asList(1, 2, -3, 4, -5, 0);
        assertEquals(4, CountNums.countNums(arr));
    }
    
    @Test
    void testMixedWithZeros() {
        List<Integer> arr = Arrays.asList(1, 2, -3, 0, 4, -5, 0);
        assertEquals(4, CountNums.countNums(arr));
    }
    
    @Test
    void testMixedWithLargeNumbers() {
        List<Integer> arr = Arrays.asList(123, 456, -789, 0, 111, -222);
        assertEquals(4, CountNums.countNums(arr));
    }
    
    @Test
    void testMixedWithSingleDigitNumbers() {
        List<Integer> arr = Arrays.asList(1, 2, -3, 4, -5, 0, 11, -22);
        assertEquals(6, CountNums.countNums(arr));
    }
    
    @Test
    void testMixedWithMultipleDigitNumbers() {
        List<Integer> arr = Arrays.asList(123, 456, -789, 0, 111, -222, 999);
        assertEquals(5, CountNums.countNums(arr));
    }
}